File: VB/MainWindowViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.ModuleInjection
Imports System.Windows.Input

Namespace DialogTest
    Public Class MainWindowViewModel
        Inherits ViewModelBase

        Public Sub New()
            ShowDialogCommand = New DelegateCommand(AddressOf OnShowDialog)
        End Sub

        Public Property ShowDialogCommand() As ICommand

        Public Async Sub OnShowDialog()
            Dim injectionResult = Await ModuleManager.DefaultWindowManager.RegisterOrShowOrActivate("DialogView", New System.Reflection.Module("MyDialogView", Function() New MyDialogViewModel(), GetType(MyDialogView)))
            System.Diagnostics.Debug.WriteLine(injectionResult.Result)
        End Sub
    End Class
End Namespace

File: CS/MainWindowViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.ModuleInjection;
using System.Windows.Input;

namespace DialogTest
{
    public class MainWindowViewModel : ViewModelBase
    {
        public MainWindowViewModel()
        {
            ShowDialogCommand = new DelegateCommand(OnShowDialog);
        }

        public ICommand ShowDialogCommand { get; set; }

        public async void OnShowDialog()
        {
            var injectionResult = await ModuleManager.DefaultWindowManager.RegisterOrShowOrActivate("DialogView",
                                           new Module("MyDialogView", () => new MyDialogViewModel(), typeof(MyDialogView)));
            System.Diagnostics.Debug.WriteLine(injectionResult.Result);
        }
    }
}
